src/CMakeLists.txt

```diff
@@ -109,15 +110,10 @@ endif ()
 #-----------------------------------------------------------------------------
 # Make the REST VOL build depend on H5pubconf.h existing
 #-----------------------------------------------------------------------------
-if (PREBUILT_HDF5_DIR)
-  add_custom_target(
-    rest_vol_pubconf_depend
-    DEPENDS ${PREBUILT_HDF5_DIR}/include/H5pubconf.h
-  )
-elseif (NOT PREBUILT_HDF5_DIR AND HDF5_FOUND)
+if (HDF5_FOUND)
   add_custom_target(
     rest_vol_pubconf_depend
-    DEPENDS ${HDF5_INCLUDE_DIRS}/H5pubconf.h
+    DEPENDS ${HDF5_BINARY_DIR}/src/H5pubconf.h
```

Comment:
I still think we need a better solution than relying on variables that may or may not exist. The REST VOL won't always be built at the same time that HDF5 is and in the case where `HDF5_FOUND` is set from a `find_package` call where HDF5 was already installed on the system, `HDF5_BINARY_DIR` isn't going to exist.

Reply:
What is the range of build options we want to support? There's building under the library with FetchContent, building the REST VOL against a prebuilt HDF5 library, downloading the HDF5 source as a submodule, finding a pre-existing (but not necessarily prebuilt) HDF5 library via `find_package`, and then whatever equivalents these methods have in autotools.

Reply:
> building the REST VOL against a prebuilt HDF5 library
> finding a pre-existing (but not necessarily prebuilt) HDF5 library via find_package

These cases should be treated as exactly the same. At a minimum, we must support building the REST VOL against an HDF5 package that is already installed on the system (manually, via spack, via a package manager, etc.) as the default and preferred building method. Anything else should be seen as alternative conveniences for different purposes.

> There's building under the library with FetchContent

Since CMakeVOL.cmake sets the CMake variables that a `find_package(HDF5)` call would otherwise set, building the REST VOL under the library via FetchContent should work automatically with no special case handling as long as the REST VOL's CMake code does the standard thing of `find_package(HDF5)` -> link to `HDF5_LIBRARIES/HDF5_C_LIBRARIES` and include `HDF5_INCLUDE_DIRS`.
That should mean that on the particular line this comment is on, keeping the custom target as depending on `HDF5_INCLUDE_DIRS/H5pubconf.h` _should_ work fine, though if there's an extra directory we're missing or something like that, we should fix that.

> downloading the HDF5 source as a submodule

Downloading the HDF5 source as a submodule should go away with this PR, so we shouldn't have to worry about that case anymore; it was just a convenience for users at the time and also for development purposes. Now that the CMake FetchContent scheme is in place, though, that should likely be the preferred method for development.

> whatever equivalents these methods have in autotools

Since Autotools is less convenient in this regard, we should consider that case as being equivalent to "building the REST VOL against an HDF5 package that is already installed on the system"

Reply:
It was my understanding the `find_package` is meant to be used with libraries that are already built/installed. I haven't been able to use it with the HDF5 source - in config mode, `find_package` tries to find the `hdf5-config.cmake` file that won't be created from `hdf5-config.cmake.in` until the HDF5 configuration step is done.  In module mode, the provided `FindHDF5.cmake` module behaves the same way when looking for the config file.

Reply:
>That should mean that on the particular line this comment is on, keeping the custom target as depending on HDF5_INCLUDE_DIRS/H5pubconf.h should work fine, though if there's an extra directory we're missing or something like that, we should fix that.

When HDF5 is built and not installed, `H5pubconf.h` is only found in `${HDF5_BINARY_DIR}/src/`, which isn't an include directory.  This becomes an issue when trying to build the REST VOL while the library is built and not installed, hence this change. We could potentially add `${HDF5_BINARY_DIR}/src/` to the list of include directories specified in `CMakeVOL.cmake`, but this seems like it will become confusing down the road.

Reply:
Yes, `find_package` is meant for looking for packages that are already installed on the system. See https://github.com/HDFGroup/hdf5/blob/develop/CMakeVOL.cmake#L150-L162 where the CMake code for building VOLs essentially rips out `find_package(HDF5)` calls from the downloaded local copy of a VOL connector. By doing this and setting the relevant CMake variables, it basically has the effect of presenting a not-yet-built HDF5 as an installed package so that the VOL can be configured, and then it builds the VOL after HDF5 gets built so that everything is available as it should be (minus the installed directories and their contents being available).

It's likely the case that we can remove the custom depends target now since IIRC it was mostly just for supporting the prebuilt HDF5 notion. Now, if you're building against a pre-installed HDF5 you don't need to depend on anything before building the VOL because HDF5 is already available. If you're using the FetchContent approach, CMake should be correctly handling the ordering so that the VOL isn't built before HDF5 is. I would try removing the custom target and see if everything builds correctly as is.

Reply:
You're right, removing the custom target didn't mess up the build. I'll set something up to handle building against not-yet-built HDF5 and update this PR.